packages/web-api-scan-runner/src/runner/runner.ts

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { convertAxeToSarif, SarifLog } from 'axe-sarif-converter';
+import { GuidGenerator } from 'common';
+import { inject, injectable } from 'inversify';
+import { isNil } from 'lodash';
+import { Logger } from 'logger';
+import { Browser } from 'puppeteer';
+import { AxeScanResults } from 'scanner';
+import { OnDemandPageScanRunResultProvider, PageScanRunReportService } from 'service-library';
+import {
+    OnDemandPageScanReport,
+    OnDemandPageScanResult,
+    OnDemandPageScanRunResult,
+    OnDemandPageScanRunState,
+    OnDemandScanResult,
+    ScanState,
+} from 'storage-documents';
+import { ScanMetadataConfig } from '../scan-metadata-config';
+import { ScannerTask } from '../tasks/scanner-task';
+import { WebDriverTask } from '../tasks/web-driver-task';
+
+@injectable()
+export class Runner {
+    constructor(
+        @inject(GuidGenerator) private readonly guidGenerator: GuidGenerator,
+        @inject(ScanMetadataConfig) private readonly scanMetadataConfig: ScanMetadataConfig,
+        @inject(ScannerTask) private readonly scannerTask: ScannerTask,
+        @inject(OnDemandPageScanRunResultProvider) private readonly onDemandPageScanRunResultProvider: OnDemandPageScanRunResultProvider,
+        @inject(WebDriverTask) private readonly webDriverTask: WebDriverTask,
+        @inject(Logger) private readonly logger: Logger,
+        @inject(PageScanRunReportService) private readonly pageScanRunReportService: PageScanRunReportService,
+        private readonly convertAxeToSarifFunc = convertAxeToSarif,
+    ) {}
+
+    public async run(): Promise<void> {
+        let browser: Browser;
+        const scanMetadata = this.scanMetadataConfig.getConfig();
+        let pageScanResult: OnDemandPageScanResult;
+
+        this.logger.logInfo(`Reading Page Scan Run id= ${scanMetadata.id}`);
+        pageScanResult = (await this.onDemandPageScanRunResultProvider.readScanRuns([scanMetadata.id]))[0];
+
+        // set scanned page run state to running
+        pageScanResult.run = this.getRunResult('running');
+
+        this.logger.logInfo(`Updating page scan to running`);
+        await this.onDemandPageScanRunResultProvider.updateScanRun(pageScanResult);
+        pageScanResult = (await this.onDemandPageScanRunResultProvider.readScanRuns([scanMetadata.id]))[0];
+
+        try {
```

Comment:
The scan can be private func

Reply:
As @devaradhanm, i would like to take the feedback in the next PR if it is perf imrpovement or code style to unblock us from doing e2e validation